ipad/ViewControllers/Watercraft Inspections/WatercraftInspectionViewController.swift

```diff
@@ -216,209 +260,421 @@ class WatercraftInspectionViewController: BaseViewController {
+                condition: highRiskDecontaminationPerformed
+                    && highRiskDecontaminationReference.isEmpty,
+                section: "inspectionOutcomes"
+            ),
+            Validation(
+                type: .decontaminationOrderNumber,
+                errorMessage: .errorDecontaminationOrderNumber,
+                condition: highRiskDecontaminationOrderIssued
+                    && highRiskDecontaminationOrderNumber <= 0,
+                section: "inspectionOutcomes"
+            ),
+            Validation(
+                type: .decontaminationOrderReason,
+                errorMessage: .errorDecontaminationOrderReason,
+                condition: highRiskDecontaminationOrderIssued
+                    && highRiskDecontaminationOrderReason.isEmpty,
+                section: "inspectionOutcomes"
+            ),
+            Validation(
+                type: .sealNumber,
+                errorMessage: .errorSealNumber,
+                condition: highRiskSealIssued
+                && highRiskSealNumber <= 0,
+                section: "inspectionOutcomes"
+            ),
+        ]
 
-                if !highRisk.decontaminationOrderIssuedInteracted {
-                    message = "\(message)\n\(counter). Please input Decontamination order issued field (Inspection Outcomes).\n"
-                    counter += 1
-                }
-                
-                // Decontamination order has been interacted with and set to "Yes", but a Record of Decontamintion number is empty
-                if highRisk.decontaminationOrderIssuedInteracted &&
-                    highRisk.decontaminationOrderIssued &&
-                    highRisk.decontaminationOrderNumber <= 0 {
-                    message = "\(message)\n\(counter). Please input the Decontamination order number (Inspection Outcomes).\n"
-                    counter += 1
+        // Iterates through the validationItems and checks conditions for any failures
+        // Aggregates the error messages into their respective arrays
+        for validation in validationItems {
+            switch validation.section {
+            case "basicInformation":
+                if isPassportHolderNewOrLaunched && validation.condition {
+                    validationErrors[0].errors.append(validation.errorMessage.rawValue)
                 }
-                
-                if highRisk.decontaminationOrderIssuedInteracted &&
-                    highRisk.decontaminationOrderIssued &&
-                    highRisk.decontaminationOrderReason.isEmpty {
-                    message = "\(message)\n\(counter). Please input the Reason for issuing a decontamination order (Inspection Outcomes).\n"
-                    counter += 1
+            case "watercraftDetails":
+                if isPassportHolderNewOrLaunched && validation.condition {
+                    validationErrors[1].errors.append(validation.errorMessage.rawValue)
```

Comment:
Should avoid the use of magic numbers for this section and use a locally scoped enum or variable

Reply:
Would add better clarity and intent, since the Array this references was declared on line 356 :)